vnext/local-cli/generator-windows/templates/cpp/src/AutolinkedNativeModules.g.h

```diff
@@ -1,17 +1,14 @@
-// AutolinkedNativeModules.g.h -- contents generated by "react-native run-windows"
-
+// AutolinkedNativeModules.g.h contents generated by "react-native autolink-windows"
+// clang-format off
 #include "pch.h"
 
 #pragma once
 
-// clang-format off
-
 namespace winrt::Microsoft::ReactNative
 {
 
 static void RegisterAutolinkedNativeModulePackages(winrt::Windows::Foundation::Collections::IVector<winrt::Microsoft::ReactNative::IReactPackageProvider> const& packageProviders)
```

Comment:
I don't think this is quite right - static in the context of a free function (even in a namespace) means it will only be visible to the translation unit it's in (i.e. it's going to be visible only to the CPP that includes this header).
in this case it's no-op because there's no code, but when you start adding code to it, you'll end up with duplicate copies of  the code. Note that you won't get duplicate definitions because its visibility is limited to the file that includes the header.

Reply:
It's only called once, in the template during app init - this originally was a macro just to insert the code into that app init location, but I was asked to make it into a function. No-one complained when I put it into the template weeks ago, and changing will require changing the template. Is there a better option?

Reply:
yeah I had noticed in the PR but forgot to comment, my bad :)
there are a few options. The right thing would be to have the declaration of the function in this header file:
```cpp
void RegisterAutolinkedNativeModulePackages(winrt::Windows::Foundation::Collections::IVector<winrt::Microsoft::ReactNative::IReactPackageProvider> const& packageProviders);
```
Then have its definition in an associated cpp file.

2nd option is to put this function into a class as a static method.

3rd less nice option is to mark the function as `__declspec(selectany)`

Reply:
This is what the cli will make for 62 apps - if I change it here then we should backport it into 0.62.

Reply:
I have a separate PR out for this fix here: https://github.com/microsoft/react-native-windows/pull/5090

That way I can get just this change backported into 62 so that apps made in 62 are set up for autolinking in 63.

Reply:
fixed.